src/ui/public/index_patterns/_field.js

```diff
@@ -40,10 +40,10 @@ export default function FieldObjectProvider(Private, shortDotsFilter, $rootScope
 
     const indexed = !!spec.indexed;
     const scripted = !!spec.scripted;
-    const sortable = spec.name === '_score' || ((indexed || scripted) && type.sortable);
-    const filterable = spec.name === '_id' || scripted || (indexed && type.filterable);
     const searchable = !!spec.searchable || scripted;
     const aggregatable = !!spec.aggregatable || scripted;
+    const sortable = spec.name === '_score' || ((indexed || scripted || aggregatable) && type.sortable);
+    const filterable = spec.name === '_id' || scripted || ((indexed || searchable) && type.filterable);
```

Comment:
Yeah, you're reading that correctly. It's been that way for quite awhile though. I'm not sure if it's intentional or just a side effect of the code evolving over time. Given that filtering and sorting are two very different operations in ES, I wouldn't be totally surprised if there's some subtlety at play here.